src/main/java/bio/terra/service/filedata/DrsIdService.java

```diff
@@ -51,12 +51,28 @@ public static DrsId fromUri(String drsUri) {
     URI uri = URI.create(drsUri);
     if (!StringUtils.equals(uri.getScheme(), "drs")
         || uri.getAuthority() == null
-        || !StringUtils.startsWith(uri.getPath(), "/")) {
+        // This is invalid for standard DRS URIs
+        || (!StringUtils.startsWith(uri.getPath(), "/") && !StringUtils.isEmpty(uri.getPath()))
+        // This is invalid for compact IDs
+        || (StringUtils.isEmpty(uri.getPath()) && !uri.getAuthority().contains(":"))) {
       throw new InvalidDrsIdException("Invalid DRS URI '" + drsUri + "'");
     }
-    String datarepoDnsName = uri.getAuthority();
-    String objectId = StringUtils.remove(uri.getPath(), '/');
-    return parseObjectId(datarepoDnsName, objectId).dnsname(uri.getAuthority()).build();
+
+    // According to the DRS spec, port numbers can't be specified so just the presence of `:` in the
```

Comment:
This inference makes me a bit nervous, but it's probably okay.

Reply:
Yeah, that's one of the things that the spec is pretty clear about so I feel ok about it but I take your point